pkg/operator/controller/gatewayapi/controller.go

```diff
@@ -60,14 +66,33 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 		}}
 	}
 
-	// watch for CRDs
-	crdPredicate := predicate.NewPredicateFuncs(func(o client.Object) bool {
-		return o.(*apiextensionsv1.CustomResourceDefinition).Spec.Group == gatewayapiv1.GroupName
-	})
+	isGatewayAPICRD := func(o client.Object) bool {
+		crd := o.(*apiextensionsv1.CustomResourceDefinition)
+		return crd.Spec.Group == gatewayapiv1.GroupName || crd.Spec.Group == experimantalGatewayAPIGroupName
+	}
+	crdPredicate := predicate.NewPredicateFuncs(isGatewayAPICRD)
 
+	// watch for CRDs
 	if err := c.Watch(source.Kind[client.Object](operatorCache, &apiextensionsv1.CustomResourceDefinition{}, handler.EnqueueRequestsFromMapFunc(toFeatureGate), crdPredicate)); err != nil {
 		return nil, err
 	}
+
+	// Index Gateway API CRDs by the "spec.group" field
+	// to enable efficient filtering during list operations.
+	// Currently, only Gateway API groups have a dedicated index field.
+	if err := mgr.GetFieldIndexer().IndexField(
```

Comment:
Could you create two indexers, one for managed and one for unmanaged CRDs?

Reply:
What would be the purpose of having two indexers?

Reply:
Ah, I see your comment on `listUnmanagedGatewayAPICRDs` now.  

Wouldn't it make more sense to do this?
```go
		client.IndexerFunc(func(o client.Object) []string {
			if isGatewayAPICRD(o) {
				return []string{"managed"}
			}
			return []string{"unmanaged"}
		})
```

and then

```go
r.cache.List(ctx, gatewayAPICRDs, client.MatchingFields{crdAPIGroupIndexFieldName: "managed"});
```
or
```go
r.cache.List(ctx, gatewayAPICRDs, client.MatchingFields{crdAPIGroupIndexFieldName: "unmanaged"});
```

Reply:
I don't quite know why we need this indexer, which picks up managed and unmanaged CRDs together, but there is code around https://github.com/openshift/cluster-ingress-operator/pull/1205/files?diff=split&w=1#r2019309675 that gets all the CRDs from this index and then iterates through to get the unmanaged ones.  One indexer for unmanaged CRDs should be enough.

Reply:
No, I guess my last comment doesn't make sense.  The index is used in order to get all Gateway API CRDs, be they managed or be they unmanaged, and then the loop in `listUnmanagedGatewayAPICRDs` checks each of those Gateway API CRDs to determine whether any unmanaged one was indexed.  This makes sense to me:  The cluster could have many CRDs, and we only index the ~half dozen (at most) that belong to Gateway API.  It's probably cheaper to iterate over a half dozen CRDs than it would be to maintain another index to try to optimize the loop away.

Reply:
I think that 2 indexes do not make sense as "managed" index would be of no use in this case.

> One indexer for unmanaged CRDs should be enough.

This may be another approach to the problem indeed. I was thinking more of the field name which matches the CRD field `spec.group` (similar to what [Miciah described here](https://github.com/openshift/cluster-ingress-operator/pull/1205#discussion_r2019502308)). I didn't respect the naming convention though.

Reply:
> One indexer for unmanaged CRDs should be enough.

My understanding from our call today is that we agreed on this approach: using one index, which has only CRDs that are _both_ in any of the API groups that belong to Gateway API _and_ not managed by cluster-ingress-operator.

Reply:
Changed the index to have a field for all unmanaged gateway API CRDs.